vision/api/Detect/Program.cs

```diff
@@ -318,6 +333,109 @@ private static object DetectDocText(Image image)
             return 0;
         }
 
+        // [START vision_async_detect_document_ocr]
+        private static object DetectDocument(string gcsSourceUri, string gcsDestinationUri)
+        {
+
+            // Supported mime_types are: 'application/pdf' and 'image/tiff'
+            string mimeType = "application/pdf";
+
+            // How many pages should be grouped into each json output file.
+            // With a file of 5 pages
+            int batchSize = 2;
+
+            var client = ImageAnnotatorClient.Create();
+
+            var feature = new Feature{
+                Type = Feature.Types.Type.DocumentTextDetection
+            };
+
+            var gcsSource = new GcsSource
+            {
+                Uri = gcsSourceUri
+            };
+
+            var inputConfig = new InputConfig { 
+                GcsSource = gcsSource,
+                MimeType = mimeType
+            };
+
+            var gcsDestination = new GcsDestination { 
+                Uri = gcsDestinationUri
+            };
+
+            var outputConfig = new OutputConfig { 
+                BatchSize = batchSize,
+                GcsDestination = gcsDestination
+            };
+
+            var asyncRequest = new AsyncAnnotateFileRequest {
+                InputConfig = inputConfig,
+                OutputConfig = outputConfig
+            };
+
+            asyncRequest.Features.Add(feature);
+
+            List<AsyncAnnotateFileRequest> requests =
+                new List<AsyncAnnotateFileRequest>();
+            requests.Add(asyncRequest);
+
+            var operation = client.AsyncBatchAnnotateFiles(requests);
+
+            Console.WriteLine("Waiting for the operation to finish");
+
+            operation.PollUntilCompleted();
+
+            // Once the rquest has completed and the output has been
+            // written to GCS, we can list all the output files.
+            var storageClient = StorageClient.Create();
+
+            Uri destinationBucket = new Uri(gcsDestinationUri, UriKind.Absolute);
+            string bucketName = destinationBucket.Host;
+            string prefix = destinationBucket.GetComponents(

```

Comment:
Pretty gnarly for a sample whose intent is to demonstrate this feature only

Consider accepting bucket name & path prefix as method parameters to make code more clear

(I realize this will make the CLI's inputs differ from Python, but I approve of this to provide more clarity)